### PR TITLE
Support SciPy1.0+

### LINKED ISSUE #680: Support SciPy 1.0
SciPy 1.0 released. It has some compatibility breaks in `scipy.sparse`. We need to fix implementations or tests.

### PR DESCRIPTION
**Pull request by okuta**
This PR supports old and new SciPy. Fix #680 .
- Accept sparse matrix in `numpy_cupy_allclose` and `numpy_cupy_array_equal`.
- Relax slicing restrictions
- Fix test to use new style test functions

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

File: cupyx/scipy/sparse/compressed.py [hunk 1]
```diff
@@ -269,13 +269,13 @@ def _get_major_slice(self, major):
             major_start += major_size
         if major_stop < 0:
             major_stop += major_size
+        major_start = max(min(major_start, major_size), 0)
```
[the remark this answers is not in the document]
Comment (okuta) [reply cupyx/scipy/sparse/compressed.py]: Yes, This PR contains `Relax slicing restrictions`.

[the remark this answers is not in the document]
Comment (unnonouno) [reply cupyx/scipy/sparse/compressed.py]: Ah, do you mean that from scipy 1.0 out of bounds index does not raise an error?

[the remark this answers is not in the document]
Comment (okuta) [reply cupyx/scipy/sparse/compressed.py]: Yes. This is same behavior as Python list and tuple slicing operation.
Previous behavior was too strict.

[the remark this answers is not in the document]
Comment (unnonouno) [reply cupyx/scipy/sparse/compressed.py]: Ok, I understood it.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**